src/main/java/io/usethesource/vallang/ITuple.java

```diff
@@ -16,6 +16,11 @@
 import io.usethesource.vallang.visitors.IValueVisitor;
 
 public interface ITuple extends Iterable<IValue>, IValue {
+    @Override
+    default int getMatchFingerprint() {
+        return 442900256 /* "tuple".hashCode() << 2 */ + arity();
```

Comment:
Mixes arity differenlty than the others?

Reply:
yes to make sure we have fewer hash collisions between INodes and ITuples. Could have chosen another prime number, but `1` is also prime.

Reply:
ok, I was just doing a final check to see if there were other misalignments.